sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1231,8 +1231,12 @@ inline std::string GetConnectionKey(std::string const& host, CurlTransportOption
   key.append(!options.SslOptions.EnableCertificateRevocationListCheck ? "1" : "0");
   key.append(options.SslVerifyPeer ? "1" : "0");
   key.append(options.NoSignal ? "1" : "0");
+  // using DefaultConnectionTimeout or 0 result in the same setting
   key.append(
-      options.ConnectionTimeout == CurlTransportOptions::DefaultConnectionTimeout ? "0" : "1");
+      (options.ConnectionTimeout == CurlTransportOptions::DefaultConnectionTimeout
+       || options.ConnectionTimeout == 0)
+          ? "0"
+          : std::to_string(options.ConnectionTimeout));
```

Comment:
I imagine this requires unit test updates, no?

Reply:
We should add some tests with different (non-default) connection timeouts.

Reply:
added